### PR TITLE
Integ 310 listen for changes in status of published content to update render of warning message

### PR DESCRIPTION
**Pull request by lilbitner**
## Purpose
This PR allows the GA4 sidebar app to listen for changes in the entry state of 'published' vs 'unpublished. This is important because we show a warning message in lieu of the app when the content is not published. If the user publishes the content, we then want to render the app appropriately. 

## Approach
The `sdk.entry` has a function called `onSysChanged` which is called when the entry sys metadata is adjusted in anyway, which includes the `publishedAt` attribute. Additionally, I made a few adjustments to assure that rendering the warning message and then subsequently rendering the app itself 'maintains' or 'clears' the appropriate state at the level of the `AnalyticsApp` component, such as the selected date range. The loading state was also adjusted to ensure the chart and any warning message render appropriately, and not before the actual gathering of data. Loading is fairly seamless now (from my local testing at least)🤞. Fixed some inconsistent margins and slight blurriness in the chart given the strict height requirement. 

## Results
https://www.loom.com/share/86238383a6f94426885156671530a4d9

## How to test
Render the GA4 app in the sidebar, and go ahead and toggle between publishing and un-publishing the entry content. This should toggle between rendering the warning message and the app itself. 

## Dependencies and/or References
[SDK docs](https://www.contentful.com/developers/docs/extensibility/app-framework/sdk/#entry)

### CODE REVIEW

File: apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx [hunk 1]
```diff
@@ -77,8 +80,10 @@ const AnalyticsApp = (props: Props) => {
 
   const metricName = runReportResponse.metricHeaders && runReportResponse.metricHeaders[0].name;
 
+  const pendingData = isEmpty(runReportResponse) && !error && runReportFetchRequirements;
```
Comment (lilbitner) [inline apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx:83]: this helps resolve the gap between when our artificial 'loading' state is done and when the data has actually been gathered.

File: apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx [hunk 2]
```diff
@@ -77,8 +80,10 @@ const AnalyticsApp = (props: Props) => {
...
+
   const renderAnalyticContent = () => {
-    if (loading) {
+    if (loading || pendingData) {
```
Comment (lilbitner) [inline apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx:86]: this helps resolve the gap between when our artificial 'loading' state is done and when the data has actually been gathered.

**Review by ghepting — APPROVED** [review]

File: apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
 import useGetFieldValue from '../useGetFieldValue';
 
 export const useSidebarSlug = (slugFieldInfo: ContentTypeValue) => {
+  const [isPublished, setIsPublished] = useState(false);
   const sdk = useSDK<SidebarExtensionSDK>();
 
   const { slugField, urlPrefix } = slugFieldInfo;
   const slugFieldValue = useGetFieldValue(slugField);
 
+  const checkPublishedStatus = (sys: ContentEntitySys) => {
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx:15]: minor nit: this is really `setPublishedStatus` isn't it?

File: apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx [hunk 4]
```diff
@@ -53,9 +56,9 @@ const AnalyticsApp = (props: Props) => {
       setLoading(false);
     }
 
-    if (reportSlug && propertyId && !isContentTypeWarning) fetchRunReportData();
+    if (runReportFetchRequirements) fetchRunReportData();
     else setTimeout(() => setLoading(false), 200);
```
Comment (mgoudy91) [inline apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx:60]: I know this isn't exactly the code you wrote here, but what is this 200 ms delay doing? seems really strange

**Review by mgoudy91 — APPROVED** [review]
This all looks good to me!

One likely unrelated question, only if you know the answer

File: apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx [hunk 5]
```diff
@@ -1,19 +1,28 @@
...
 import useGetFieldValue from '../useGetFieldValue';
 
 export const useSidebarSlug = (slugFieldInfo: ContentTypeValue) => {
+  const [isPublished, setIsPublished] = useState(false);
   const sdk = useSDK<SidebarExtensionSDK>();
 
   const { slugField, urlPrefix } = slugFieldInfo;
   const slugFieldValue = useGetFieldValue(slugField);
 
+  const setPublishedStatus = (sys: ContentEntitySys) => {
```
Comment (ryunsong-contentful) [inline apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx:15]: :nit `handlePublishedStatus`

File: apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx [hunk 6]
```diff
@@ -53,9 +56,9 @@ const AnalyticsApp = (props: Props) => {
       setLoading(false);
     }
 
-    if (reportSlug && propertyId && !isContentTypeWarning) fetchRunReportData();
-    else setTimeout(() => setLoading(false), 200);
-  }, [api, reportRequestParams, reportSlug, propertyId, isContentTypeWarning]);
+    if (runReportFetchRequirements) fetchRunReportData();
+    else setLoading(false);
```
Comment (ryunsong-contentful) [inline apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx:60]: nit: You can just remove line 56 and remove this else and just have it `setLoading(false)` everytime

**Review by ryunsong-contentful — APPROVED** [review]

File: apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx (hunk 5 above)
Comment (lilbitner) [reply apps/google-analytics-4/frontend/src/hooks/useSidebarSlug/useSidebarSlug.tsx]: adjusted!

File: apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx (hunk 6 above)
Comment (lilbitner) [reply apps/google-analytics-4/frontend/src/components/main-app/AnalyticsApp/AnalyticsApp.tsx]: nice, great catch, have adjusted 💯

**Pull request merged**